### PR TITLE
Nci s2 c3 sync prod

### PR DESCRIPTION
**Pull request by JM-GA**
PR for performing the NCI to AWS Sync for S2.
Before we run in prod we need to:
- deploy updated SNS indexing code to prod (Jonno)
- test that maturity archiving when 'final' arrives is correctly being handled in dev (James)
- raise a PR in datakube to increase the user permissions to allow multi-part copy (James)

### CODE REVIEW

**Review by uchchwhash — DISMISSED** [review]
No objections from me

File: scripts/nci_c3_s2_data_verification_aws.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import boto3
+import os
+from botocore.errorfactory import ClientError
+from botocore import UNSIGNED
+from botocore.config import Config
+
+
+s3_bucket = 'dea-public-data-dev'
+product_path = 'baseline'
+nci_directory = '///g/data/ka08/ga'
```
Comment (jmettes) [inline scripts/nci_c3_s2_data_verification_aws.py:16]: Just curious, why are there multiple slashes?

File: dags/collection3/nci_c3_s2_sync.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            export AWS_SECRET_ACCESS_KEY={{aws_creds.secret_key}}
+            time ~/bin/s5cmd --stat --numworkers 20 run {{ work_dir }}/missing_file.txt
+            """
+        ),
+        remote_host="gadi-dm.nci.org.au",
+        params={"aws_hook": aws_hook},
+        timeout=10 * HOURS,
+    )
+
+    upload_uploader_script >> upload_data_verification
```
Comment (jmettes) [inline dags/collection3/nci_c3_s2_sync.py:206]: I might not know airflow enough to understand, but what's the difference between this and doing:

```
upload_uploader_script >> upload_data_verification >> generate_list_of_s2_to_upload >> generate_list_of_s2_to_upload >> generate_commands >> execute_upload >> data_verification_check >> execute_missing_files_upload
```

If so is there any reason not to do this? Like parallelism or something?

Comment (JM-GA) [reply dags/collection3/nci_c3_s2_sync.py:206]: I believe the logic is the same re dependency flow between the two - just a different style of presenting it.

File: scripts/nci_c3_s2_data_verification_aws.py (hunk 1 above)
Comment (JM-GA) [reply scripts/nci_c3_s2_data_verification_aws.py:16]: yeah they can be removed - no reason to have them,

**Review by jmettes — APPROVED** [review]

**Pull request merged**